src/Microsoft.AspNet.Authentication.OAuth/OAuthAuthenticationHandler.cs

```diff
@@ -159,14 +160,44 @@ protected override AuthenticationTicket AuthenticateCore()
             return new TokenResponse(oauth2Token);
         }
 
-        protected virtual async Task<AuthenticationTicket> GetUserInformationAsync(AuthenticationProperties properties, TokenResponse tokens)
+        protected virtual async Task<AuthenticationTicket> CreateTicketAsync(AuthenticationProperties properties, TokenResponse tokens)
         {
-            var context = new OAuthGetUserInformationContext(Context, Options, Backchannel, tokens)
+            var identity = new ClaimsIdentity(Options.ClaimsIssuer);
+
+            if (Options.SaveTokens)
             {
-                Properties = properties,
+                identity.AddClaim(new Claim("access_token", tokens.AccessToken, ClaimValueTypes.String, Options.ClaimsIssuer));
```

Comment:
No, it wasn't :smile: 

Should we create a new setting per claim (`access_token`, `expires_in`, `refresh_token` and `token_type`) or should we use a more elaborated claims map dictionary?